litmus-portal/graphql-server/pkg/handlers/file_handler.go

```diff
@@ -53,45 +60,48 @@ func FileHandler(w http.ResponseWriter, r *http.Request) {
 }
 
 func GetManifest(token string) ([]byte, int, error) {
-
-	id, err := cluster.ClusterValidateJWT(token)
+	clusterID, err := cluster.ClusterValidateJWT(token)
 	if err != nil {
 		return nil, 404, err
 	}
 
-	reqCluster, err := dbOperationsCluster.GetCluster(id)
+	reqCluster, err := dbOperationsCluster.GetCluster(clusterID)
 	if err != nil {
 		return nil, 500, err
 	}
 
-	scope := os.Getenv("PORTAL_SCOPE")
-	if scope == clusterScope {
+	if scope == namespaceScope && endpoint == "" {
+		return nil, 500, errors.New("CHAOS_CENTER_UI_ENDPOINT env is empty for namespace scope installation")
+	}
+
+	if endpoint != "" {
+		subscriberConfiguration.GQLServerURI = endpoint + "/query"
```

Comment:
If user is providing UI url, then appending `/query` might not work, right? According to nginx, it will be available `/ws/query`, `/query` will only work for NodePort.

Reply:
yes it needs to be /ws/query as we are asking for the UI endpoint, even for loadbalancer or nodeport it needs to be /ws/query because we are always asking for UI endpoint and not graphql server endpoint